### PR TITLE
Feat: project and project Details implements

### PR DESCRIPTION
**Pull request by wangbow**
### Description of your changes

Fixes # project and project Details implements

I have: 

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [x] Run `yarn lint` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: src/interface/project.ts [hunk 1]
```diff
@@ -4,4 +4,10 @@ export interface Project {
   description?: string;
   createTime?: string;
   updateTime?: string;
+  owner?: Owner;
```
Comment (barnettZQG) [inline src/interface/project.ts:7]:
```suggestion
  owner?: NameAlias;
```

File: src/interface/target.ts [hunk 2]
```diff
@@ -9,4 +10,5 @@ export type Target = {
     namespace?: string;
   };
   variable?: any;
+  project?: Project;
```
Comment (barnettZQG) [inline src/interface/target.ts:13]:
```suggestion
  project?: NameAlias;
```

File: src/api/project.ts [hunk 3]
```diff
@@ -7,10 +8,30 @@ const baseURLOject = getDomain();
...
 const url = isMock ? project_mock : project;
 
-export function getProjectList(params: any) {
-  return get(url, params).then((res) => res);
+export function getProjectList(params: { page?: number; pageSize?: number }) {
+  return get(url, { params: params }).then((res) => res);
 }
 
-export function createProject(params: any) {
+export function createProject(params: Project) {
```
Comment (barnettZQG) [inline src/api/project.ts:15]: Please redefine the interface for CreateProject and UpdatePorject, Their parameters are different from Project

File: src/api/project.ts [hunk 4]
```diff
@@ -7,10 +8,30 @@ const baseURLOject = getDomain();
...
+  const urlPath = project + `/${params.name}`;
+  return rdelete(urlPath, {}).then((res) => res);
+}
+
+export function getProjectDetail(params: { projectsName: string }) {
+  const urlPath = project + `/${params.projectsName}`;
+  return get(urlPath, {}).then((res) => res);
+}
+
+export function getProjectDetailTargets(params: { projectsName: string }) {
```
Comment (barnettZQG) [inline src/api/project.ts:34]:
```suggestion
export function getProjectTargetList(params: { projectName: string }) {
```

File: src/layout/Content/index.tsx [hunk 5]
```diff
@@ -122,6 +128,62 @@ export default function Content() {
       <Route path="/addons/:addonName" component={Addons} />
       <Route path="/addons" component={Addons} />
       <Route path="/users" component={Users} />
+      <Route exact path="/projects" component={Projects} />
+      <Route
+        exact
+        path="/projects/:projectsName"
```
Comment (barnettZQG) [inline src/layout/Content/index.tsx:134]: Please change all `projectsName` to `projectName`

File: src/model/project.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { getProjectDetail } from '../api/project';
+import type { Project } from '../interface/project';
+
+interface projectDetailState {
+  projectDetails: {};
```
Comment (barnettZQG) [inline src/model/project.ts:5]: Explicitly define the type.

File: src/interface/project.ts (hunk 1 above)
Comment (wangbow) [reply src/interface/project.ts]: done.

File: src/interface/target.ts (hunk 2 above)
Comment (wangbow) [reply src/interface/target.ts]: done.

File: src/api/project.ts (hunk 3 above)
Comment (wangbow) [reply src/api/project.ts]: done.

File: src/api/project.ts (hunk 4 above)
Comment (wangbow) [reply src/api/project.ts]: done.done.

File: src/layout/Content/index.tsx (hunk 5 above)
Comment (wangbow) [reply src/layout/Content/index.tsx]: done.

File: src/model/project.ts (hunk 6 above)
Comment (wangbow) [reply src/model/project.ts]: done.

**Review by barnettZQG — APPROVED** [review]
lgtm

**Pull request merged**